Update robotest upgrade targets.

wadells opened the pull request:
## Description
This patch drops validation of 6.1 to 7.1.0-alpha upgrades.
These are replaced with validation from 7.0. Furthermore 6.2 and 6.3
are included per our <=2 minor kubernetes version skew policy and some
recent discussion with @knisbet about whether we support non-LTS to
non-LTS upgrades.

## Type of change
* Internal change (not necessarily a bug fix or a new feature)

## TODOs
<!--Required. Keep only those that apply and check them off as they get completed.-->

- [x] Self-review the change
- [x] Perform testing
- [x] Address review feedback

## Testing done
The PR build is sufficient testing.

wadells committed:
Update robotest upgrade targets.

This patch drops validation of 6.1 to 7.1.0-alpha upgrades.
These are replaced with validation from 7.0. Furthermore 6.2 and 6.3
are included per our <=2 minor kubernetes version skew policy and some
recent discussion with @knisbet about whether we support non-LTS to
non-LTS upgrades.

knisbet reviewed (approved):
Overall, I think this reflects our model and the underlying kubernetes version skew, so good stuff.

The only thing we may want to consider, as an exception to the underlying model, is 6.2 has been out of support for more than 6 months at this point, and 6.3 hasn't had much adoption, with enterprise customers going to 7.0, so it might be safe to try and save a few CPU cycles, dropping 6.2 and maybe 6.3 for the time being.

wadells committed:
Remove 6.2 and 6.3 from PR builds.

Quoting @knisbet:

"6.2 has been out of support for more than 6 months at this point,
and 6.3 hasn't had much adoption, with enterprise customers going to
7.0, so it might be safe to try and save a few CPU cycles, dropping
6.2 and maybe 6.3 for the time being."

I'm keeping 6.2 and 6.3 in the nightlies.

wadells commented:
> The only thing we may want to consider, as an exception to the underlying model, is 6.2 has been out of support for more than 6 months at this point, and 6.3 hasn't had much adoption, with enterprise customers going to 7.0, so it might be safe to try and save a few CPU cycles, dropping 6.2 and maybe 6.3 for the time being.

I yanked 6.2 and 6.3 from PR builds in 5269615, but I left them in nightly builds.  This is probably an appropriate level of testing for "technically supported but unlikely to ever be used"
